Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.cpp

```diff
@@ -142,12 +143,10 @@ class RemoteAudioDestination final
             status = noErr;
         }
 
-        unsigned requestedSamplesCount = m_extraRequestedFrames;
-        for (; requestedSamplesCount < numberOfFrames; requestedSamplesCount += WebCore::AudioUtilities::renderQuantumSize) {
-            // Ask the audio thread in the WebContent process to render a quantum.
-            m_renderSemaphore.signal();
-        }
-        m_extraRequestedFrames = requestedSamplesCount - numberOfFrames;
+        static_assert(std::atomic<UInt32>::is_always_lock_free, "Shared memory atomic usage assumes lock free primitives are used");
+        RELEASE_ASSERT(m_frameCount->size() == sizeof(numberOfFrames));
+        WTF::atomicExchangeAdd(static_cast<uint32_t*>(m_frameCount->data()), numberOfFrames);
```

Comment:
This ends up casting the SharedMemory pointer into a `WTF::Atomics<uint32_t>*`, which wraps a `std::atomics<uint32_t>`. Your SharedMemory has `size(uint32_t)`. What guarantees that this is the same size as `std::atomics<uint32_t>`? As far as I know there is no such guarantee so this usually is the case.

Reply:
This is a good point, I updated the PR to allocate sizeof(atomic<uint32_t>) shared memory.

We should probably also update `atomicExchangeAdd` and similar functions with static asserts, that is outside the scope of this PR, I can file an issue tracking it